browsergym/experiments/src/browsergym/experiments/loop.py

```diff
@@ -412,17 +419,22 @@ def make_stats(self):
 
         self.stats = stats
 
-    def save_step_info(self, exp_dir, save_json=False, save_jpg=True):
+    def save_step_info(self, exp_dir, save_json=False, save_jpg=True, save_som=False):
+
+        screenshot = self.obs.pop("screenshot", None)
+        screenshot_som = self.obs.pop("screenshot_som", None)
+
+        if save_jpg and screenshot is not None:
+            img = Image.fromarray(screenshot)
+            img.save(exp_dir / f"screenshot_step_{self.step}.jpg")
```

Comment:
JPG will incur some loss in the data. I'd rather make sure the images are saved using lossless compression, like PNG

Reply:
perhaps we should not pop on self.obs, but a copy of it. Otherwise, this could
have unintended consequence. e.g. if (in the future) we try to get the image
from obs in step_info after save_step_info, but it's not there.

Reply:
@max, I'm not sure about lossless, we want leaner traces in order to build a big
dataset. I'm not sure that llms are really sensitive to lossy (assuming good
quality compression)

Reply:
i need to pop it so it's not saved in the pkl file, i could just add it back after pickling.
or maybe i dont understand your solution

Reply:
I was thinking of making a copy of obs before popping, but I didn't think it through. 
pop, save, and add back is probably better.

Reply:
👍